### ISSUE TITLE
[MASST] -Feature request - variable m/z range for analog search

### ISSUE THREAD

**Issue by RaphaelR87** [CCMS-UCSD/GNPS_Workflows#496]
Hi everyone,

as MASST is about to become the BLAST analog for mass-spec based metabolomics, I think the analog search is of utmost value.
However, so far the analog search is limited to 100 Da max. While that allows for detecting many common biotransformation reactions of metabolites in the human body some very common biotransformations to 'detoxify' cannot not be detected such as conjugates of glucuronic acid and glutathion.

Furthermore, Natural Product Chemists would love to see analogs of their compounds of interest for discovery purposes but also to address envinronmental/ecological/evolutionary questions.
The limit of 100 Da limits the usefulness of MASST for peptidic natural products dramatically (others of course too that have aglycon-glycoside pairs) as many amino acids are > 100 Da

Thanks for considering, it is a great tool already

Best,
Raphael

**Comment by lfnothias** [comment]
Ming will probably provide a more comprehensive answer.
- The challenge is increasing to more 100 Da will be extremely slow and unsustainable. 
- I would rather recommend to run molecular network with all the datasets that matched by MASST search. That feature is already available.

An other trick would be change the precursor ion mass value in search, but I don't recommend that ^^

**Comment by RaphaelR87** [comment]
Yeah, I was afraid that increasing computing time might have been the issue.

- ok so the re-analyze button will include all query-matching 'mzML files' and not only the 'specific scans of a file'? That is a great temporary solution for many cases (not in case a strain produces a very close analog of your compound, but not your original compound, which is quite commonly found)
- I still think a wider m/z range that can be pre-selected by the user would be very valuable. One of the most valuable properties of BLAST is that you can deduce phylogenetic relationships by comparing similar (not identical) proteins. And if a m/z range of 250Da would increase my analysis time from 30 min to let us a couple hours, it is totally worth it in my opinion
- thanks for the 'trick' I will give it a try

**Comment by lfnothias** [comment]
Do you know what modifications you are looking for ? or untargeted?

**Comment by RaphaelR87** [comment]
These are general suggestions.
But in one case I am looking for analogs of a known and GNPS-annotated NRP. So modifications I am looking for amongst others are compounds with additional (non-canonical) AAs

**Comment by lfnothias** [comment]
So you expect multiple modifications with non-canonical AAs ?

**Comment by justinjjvanderhooft** [comment]
Some (most) non-canonical amino acids are > 100 Da....

**Comment by RaphaelR87** [comment]
exactly, that was the point I wanted to make (thanks @justinjjvanderhooft). Actually I think that applies also to all canonical amino acids besides glycine and alanine.
Following this thought in my opinion it is not just a nice feature to have, but more of a crucial parameter if you want to compare it to -and make MASST as successful as BLAST.

Do not get me wrong MASST is a fantastic idea and it is already great in its current status.

**Comment by lfnothias** [comment]
Thanks a lot for adding your input @justinjjvanderhooft 

@RaphaelR87 , when you said "These are general suggestions." Did you actually meant "general modifications" !?

**Comment by lfnothias** [comment]
I think this is a very good case for the MASST prototype we discussed a while ago with @mwang87 and @XsirdanielX. 
@RaphaelR87 do you wanna help and test it by providing a list for the modifications you are interested in of: delta m/z expected / molecular formula for difference / name for these possible AA modifications ? Note that we already have simple general modifications (like oxidation etc), so no need to include them.

**Comment by RaphaelR87** [comment]
@lfnothias 
Haha sorry what I wanted to say was that my suggestions are of general nature, meaning everyone could benefit from a m/z range defined by the user no matter if the research question is used for targeted or untargeted metabolomics.

**Comment by RaphaelR87** [comment]
sure I can write a list, but that could become very extensive

**Comment by justinjjvanderhooft** [comment]
Glycosylatios (162, 146, 132), glucuronidations (178), and some other interesting modifcations are all >100. Typically, when I use analogue search, I put it on 200….

**Comment by lfnothias** [comment]
@RaphaelR87 it is fine, you can provide a first non-extensive version just to test

**Comment by justinjjvanderhooft** [comment]
@RaphaelR87 - the table in the supporting information of this paper could be good starting point: https://www.nature.com/articles/nchembio.684 <https://www.nature.com/articles/nchembio.684>

**Comment by lfnothias** [comment]
Excellent @justinjjvanderhooft 
@RaphaelR87 no need for MF delta. Just m/z delta and name is fine.

**Comment by justinjjvanderhooft** [comment]
Whilst we are at it: https://pubmed.ncbi.nlm.nih.gov/24191063/ - also
interesting mz deltas for saccharide moieties!

**Comment by RaphaelR87** [comment]
Great, yeah I was hoping that a list must already exists. Probably there will be an updated one from the Dereplicator paper and/or Norine database. I will check.

**Comment by RaphaelR87** [comment]
[Amino_acids_delta_m_z_Name.txt](https://github.com/CCMS-UCSD/GNPS_Workflows/files/4519042/Amino_acids_delta_m_z_Name.txt)

**Comment by RaphaelR87** [comment]
Norine DB has 543 monomer entries which are listed here: https://bioinfo.cristal.univ-lille.fr/norine/rest/monomers
There probably is a way to get the delta m/z for these residues too in an automated fashion.

**Comment by lfnothias** [comment]
Lets ask @alexeigurevich @hoseinmohimani
